dags/iop/sftp_service.py

```diff
@@ -8,7 +11,35 @@ def __init__(self):
         super().__init__(
             host=os.getenv("IOP_FTP_HOST", "localhost"),
             username=os.getenv("IOP_FTP_USERNAME", "airflow"),
-            password=os.getenv("IOP_FTP_PASSWORD", "airflow"),
+            password=None,
             port=int(os.getenv("IOP_FTP_PORT", "2222")),
             dir=os.getenv("IOP_FTP_DIR", "upload/iop"),
         )
+        self.private_key_content = os.getenv("IOP_FTP_SSH_PRIVATE_KEY")
+
+    def _SFTPService__connect(self):
```

Comment:
I would move the logic in the SFTPService to be reusable in other publishers if needed. If it has an SSH key, connect with it; otherwise, connect with a password. WDYT?